### DISCUSSION TITLE
persistent history

### DISCUSSION THREAD

**Discussion by jackielii** [gokcehan/lf#1145] [Ideas]
I think this is a debatable one. When I saw the option `history` in the doc I thought I can turn on `cd` history, turns out it's command history. Wouldn't it be better to be called `cmdhistory on/off`?

My use case is I keep pressing `q` accidently and want to relaunch lf and quickly jump back to last direction I was on. I press `[` but realised the `'` register is gone.

I wished lf had persistent marks/registers like vim, where I can just jump back to where I was after I closed it.

One potential problem I can think of is multiple instances would race to insert into marks. But I don't think that's a real problem.

Fortunately I found this hack:

```
cmd on-quit ${{
    printf "':%s\n" $f >> $HOME/.local/share/lf/marks
}}
```

**Comment by jackielii** [comment]
FYI, for those who wants a more reliable solution (the original keep appending new `'` register):

```
cmd on-quit ${{
    # replace or append ' register
    marks=$HOME/.local/share/lf/marks
    grep -q -e "^':.*$" $marks
    if [ $? -eq 0 ]; then
        # BSD sed doesn't support -i without an extension, remove '' if using gnu sed
        sed -i '' "/^':.*$/d" $marks
    fi
    echo "':${f%/*}" >> $marks
}}

```

**Comment by joelim-work** [comment]
Even simpler is to just unconditionally delete every line starting with a `'`:

```
cmd on-quit ${{
    marks=~/.local/share/lf/marks
    sed -i "/^'/d" "$marks"
    echo "':$PWD" >> "$marks"
}}
```
